### PR TITLE
Do not show profile menu item for anonymous user

### PR DESCRIPTION
**Pull request by sharvaniharan**
**Phab:** https://phabricator.wikimedia.org/T269447

### CODE REVIEW

File: app/src/main/java/org/wikipedia/dataclient/mwapi/MwQueryPage.java [hunk 1]
```diff
@@ -193,6 +194,9 @@ public long getParentRevId() {
         public String getUser() {
             return StringUtils.defaultString(user);
         }
+        public boolean isAnon() {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/dataclient/mwapi/MwQueryPage.java:197]: Please add an empty between methods.

File: app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt [hunk 2]
```diff
@@ -428,8 +428,15 @@ class ArticleEditDetailsFragment : Fragment(), WatchlistExpiryDialog.Callback, L
...
     override fun onPrepareOptionsMenu(menu: Menu) {
         super.onPrepareOptionsMenu(menu)
-        menu.findItem(R.id.menu_user_profile_page)?.title = getString(R.string.menu_option_user_profile, currentRevision?.user)
-        menu.findItem(R.id.menu_user_contributions_page)?.title = getString(R.string.menu_option_user_contributions, currentRevision?.user)
+        val userProfileMenuItem = menu.findItem(R.id.menu_user_profile_page)
+        currentRevision?.let {
+            if (it.isAnon) {
+                userProfileMenuItem.isVisible = false
+            } else {
+                userProfileMenuItem?.title = getString(R.string.menu_option_user_profile, it.user)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt:436]: Not sure why adding `?` after `userProfileMenuItem`.

File: app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt [hunk 3]
```diff
@@ -428,8 +428,15 @@ class ArticleEditDetailsFragment : Fragment(), WatchlistExpiryDialog.Callback, L
...
+            }
+            menu.findItem(R.id.menu_user_contributions_page)?.title = getString(R.string.menu_option_user_contributions, it.user)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt:438]: Same as above.

**Review by cooltey — CHANGES_REQUESTED** [review]

**Comment by sharvaniharan** [comment]
Self-merging this, so Robin can test this in his time, please feel free to leave more comments if any, and I will address in a separate PR

**Pull request merged**